qiita_db/user.py

```diff
@@ -441,6 +445,18 @@ def shared_analyses(self):
             TRN.add(sql, [self._id, qiita_config.portal])
             return set(TRN.execute_fetchflatten())
 
+    @property
+    def unread_messages(self):
+        """Returns all unread messages for a user"""
+        with TRN:
+            sql = """SELECT message_id, message, message_time, read
+                     FROM qiita.message_user
+                     JOIN qiita.message USING (message_id)
+                     WHERE email = %s AND read = FALSE
+                     ORDER BY message_time DESC"""
+            TRN.add(sql, [self._id])
+            return TRN.execute_fetchindex(-1)
```

Comment:
The -1 is not needed, remove for consistency with other codebase and for magic numbers.